src/plugins/console/public/services/autocomplete.ts

```diff
@@ -28,53 +54,165 @@ export class AutocompleteInfo {
     }
   }
 
-  public retrieve(settings: Settings, settingsToRetrieve: DevToolsSettings['autocomplete']) {
+  public retrieve(settings: Settings, settingsToRetrieve: SettingsToRetrieve) {
     this.clearSubscriptions();
-    this.http
-      .get<MappingsApiResponse>(`${API_BASE_PATH}/autocomplete_entities`, {
-        query: { ...settingsToRetrieve },
-      })
-      .then((data) => {
-        this.load(data);
-        // Schedule next request.
-        this.pollTimeoutId = setTimeout(() => {
-          // This looks strange/inefficient, but it ensures correct behavior because we don't want to send
-          // a scheduled request if the user turns off polling.
-          if (settings.getPolling()) {
-            this.retrieve(settings, settings.getAutocomplete());
-          }
-        }, settings.getPollInterval());
+    const templateSettingsToRetrieve = {
+      ...settingsToRetrieve,
+      legacyTemplates: settingsToRetrieve.templates,
+      indexTemplates: settingsToRetrieve.templates,
+      componentTemplates: settingsToRetrieve.templates,
+    };
+
+    Promise.allSettled([
+      this.retrieveMappings(settingsToRetrieve),
+      this.retrieveAliases(settingsToRetrieve),
+      this.retrieveDataStreams(settingsToRetrieve),
+      this.retrieveTemplates(templateSettingsToRetrieve),
+    ]).then(() => {
+      this.pollTimeoutId = setTimeout(() => {
+        // This looks strange/inefficient, but it ensures correct behavior because we don't want to send
+        // a scheduled request if the user turns off polling.
+        if (settings.getPolling()) {
+          this.retrieve(settings, settings.getAutocomplete());
+        }
+      }, settings.getPollInterval());
+    });
+  }
+
+  private retrieveSettings<T>(
+    settingsKey: SettingsKey,
+    settingsToRetrieve: SettingsToRetrieve
+  ): Promise<T | {}> {
+    const settingKeyToPathMap = {
+      fields: '_mapping',
+      indices: '_aliases',
+      legacyTemplates: '_template',
+      indexTemplates: '_index_template',
+      componentTemplates: '_component_template',
+      dataStreams: '_data_stream',
+    };
+
+    // Fetch autocomplete info if setting is enabled and if user has made changes.
+    if (settingsToRetrieve[settingsKey]) {
+      return send({
+        http: this.http,
+        method: 'GET',
+        path: `${settingKeyToPathMap[settingsKey]}?pretty=false`, // pretty=false to compress the response and save bandwidth by avoiding whitespace
+        asSystemRequest: true,
       });
+    } else {
+      // If the user doesn't want autocomplete suggestions, then clear any that exist
+      return Promise.resolve({});
+    }
+  }
+
+  private async retrieveMappings(settingsToRetrieve: SettingsToRetrieve) {
+    const mappings = await this.retrieveSettings<IndicesGetMappingResponse>(
+      ENTITIES.FIELDS,
+      settingsToRetrieve
+    );
+
+    if (mappings) {
+      const maxMappingSize = Object.keys(mappings).length > 10 * 1024 * 1024;
```

Comment:
Could you please explain? Looks like we already loaded all `mappings` here. 
It seems strange to me to upload 10MB of data to the client, and then show an error message and throw all this data away

Reply:
Addressed with https://github.com/elastic/kibana/pull/140569/commits/db281e37bf4177fce39e39b00d2502052a3e1fe9 as we discussed offline